src/app/games/martian-chess/tests/MartianChessDummyMinimax.spec.ts

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable max-lines-per-function */
+import { Coord } from 'src/app/jscaip/Coord';
+import { Player } from 'src/app/jscaip/Player';
+import { RulesUtils } from 'src/app/jscaip/tests/RulesUtils.spec';
+import { Table } from 'src/app/utils/ArrayUtils';
+import { MGPMap } from 'src/app/utils/MGPMap';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MartianChessDummyMinimax } from '../MartianChessDummyMinimax';
+import { MartianChessMove } from '../MartianChessMove';
+import { MartianChessNode, MartianChessRules } from '../MartianChessRules';
+import { MartianChessCapture, MartianChessState } from '../MartianChessState';
+import { MartianChessPiece } from '../MartianChessPiece';
+
+describe('MartianChessDummyMinimax', () => {
+
+    const _: MartianChessPiece = MartianChessPiece.EMPTY;
+    const A: MartianChessPiece = MartianChessPiece.PAWN;
+    const B: MartianChessPiece = MartianChessPiece.DRONE;
+    const C: MartianChessPiece = MartianChessPiece.QUEEN;
+
+    let rules: MartianChessRules;
+    let minimax: MartianChessDummyMinimax;
+
+    function isPawnMove(move: MartianChessMove, state: MartianChessState): boolean {
+        return state.getPieceAt(move.coord) === MartianChessPiece.PAWN;
+    }
+    function isDroneMove(move: MartianChessMove, state: MartianChessState): boolean {
+        return state.getPieceAt(move.coord) === MartianChessPiece.DRONE;
+    }
+    beforeEach(() => {
+        rules = new MartianChessRules(MartianChessState);
+        minimax = new MartianChessDummyMinimax(rules, 'MartianChessDummyMinimax');
+    });
+    it('should includes all moves at first turn (but no call the clock)', () => {
+        // Given the initial state
+        const state: MartianChessState = MartianChessState.getInitialState();
+        const node: MartianChessNode = new MartianChessNode(state);
+
+        // When asking the list of moves
+        const moves: MartianChessMove[] = minimax.getListMoves(node);
+
+        // Then half the moves should include call the clock
+        const notCalled: MartianChessMove[] = moves.filter((m: MartianChessMove) => m.calledTheClock === false);
+        expect(notCalled.length).toBe(13);
```

Comment:
// la moitié des mouvements doivent call the clock
expect(13 mouvements (= tous) ne font pas call the clock)